torch/nested/_internal/nested_tensor.py

```diff
@@ -189,6 +165,10 @@ def __tensor_unflatten__(inner_tensors: Dict, meta):
             requires_grad=meta["requires_grad"],
         )
 
+    def __tensor_mark_dynamic__(self, outer_policy):
+        # collapsed ragged dim must always be dynamic
+        torch._dynamo.mark_dynamic(self._values, self._ragged_idx - 1)
```

Comment:
only necessary for now because singleton symints are expected to be symbolicized and don't work downstream unless they are